### PR TITLE
Allow moving a branch to an ancestor of its head.

### LINKED ISSUE #6367: Moving branch head to parent of current branch head errors
We are getting what looks like a collection error when moving the branch head to the parent of the current branch head. The error is: `commits split.user@master=47edacb94b914bb7aaa73e1c1f9d2aff already exists`.

### PR DESCRIPTION
**Pull request by PFedak**
Fixes #6367

### CODE REVIEW

**Comment by PFedak** [comment]
Fix #6367

**Review by Tryneus — CHANGES_REQUESTED** [review]

File: src/internal/transactionenv/env.go [hunk 1]
```diff
@@ -268,20 +287,28 @@ func (env *TransactionEnv) WithTransaction(ctx context.Context, cb func(Transact
...
-		directTxn := NewDirectTransaction(env, txnCtx)
-		if overrideID != nil {
-			id, err := overrideID(txnCtx)
-			if err != nil {
-				return err
+	useOverride := overrideID != nil
+	for {
+		if err := env.WithWriteContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
+			directTxn := NewDirectTransaction(env, txnCtx)
+			if useOverride && overrideID != nil {
```
[the remark this answers is not in the document]
Comment (PFedak) [reply src/internal/transactionenv/env.go:294]: Ah of course, good point.

File: src/internal/transactionenv/env.go [hunk 2]
```diff
@@ -249,6 +251,23 @@ func (t *appendTransaction) DeleteRoleBinding(original *auth.Resource) error {
...
+
+func (e ErrInconsistentCommit) Is(other error) bool {
+	_, ok := other.(ErrInconsistentCommit)
+	return ok
+}
+
+func (e ErrInconsistentCommit) Error() string {
+	return fmt.Sprintf("inconsistent dependencies: cannot create commit from %s - branch (%s) already has a commit in this transaction", e.Commit, e.Branch.Name)
+}
+
```
[the remark this answers is not in the document]
Comment (PFedak) [reply src/internal/transactionenv/env.go]: No, I was thinking it was kind of "about" transactions, but you're right that it's really a PFS level thing. I'll move it back

**Comment by Tryneus** [comment]
LGTM

**Review by Tryneus — APPROVED** [review]

**Pull request merged**